pkg/proxy/ipvs/proxier.go

```diff
@@ -1688,7 +1765,15 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 		curEndpoints.Insert(des.String())
 	}
 
-	for _, epInfo := range proxier.endpointsMap[svcPortName] {
+	endpoints := proxier.endpointsMap[svcPortName]
+
+	if !onlyNodeLocalEndpoints && utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TopologyAwareServiceRouting) {
+		proxier.mu.Lock()
```

Comment:
Looks like `proxier.mu` should be a `sync.RWMutex` since this will be concurrent read

Reply:
and line above can use `proxy.mu.RLock()`

Reply:
Oh sorry, I didn't notice that before, it is a big mistake.

Reply:
I think we can just remove the lock here, and we may not need `sync.RWMutex` at this point, if we `retrieving a read lock at the beginning of syncProxyRules`,  it might cause concurrent read and write when `syncProxyRules` is called concurrently (e.g. serviceMap and endpointsMap will be both updated and read in `syncProxyRules`)